packages/generator-ds/src/component/index.ts

```diff
@@ -58,40 +62,14 @@ export default class ComponentGenerator extends Generator<ComponentGeneratorOpti
     };
   }
 
-  /**
-   * Gets the path to the component's directory relative to the current working directory.
-   * Pascal-cases the final directory name to match React component naming conventions.
-   * @param inPath - The path to resolve, relative to the current working directory
-   * @return Path to the component's directory relative to the current working directory
-   * @example
-   * destinationPath("path/to/button/index.ts") // => "/path/to/Button/index.ts"
-   */
-  destinationPath(...inPath: string[]): string {
-    const rawPath = super.destinationPath(...inPath);
-    const dirName = path.dirname(rawPath);
-
-    let componentFolder = path.resolve(dirName);
-
-    // Replace the last segment of the path with the Pascal-cased version
-    if (inPath.includes("/")) {
-      componentFolder = path.resolve(
-        componentFolder,
-        "..",
-        casing.toPascalCase(path.basename(dirName)),
-      );
-    }
-
-    // Append the original file name to the new path
-    const fileName = path.basename(rawPath);
-
-    return path.join(componentFolder, fileName);
-  }
-
   writing(): void {
     if (!this.answers) return;
 
-    const componentName = casing.toPascalCase(
-      path.basename(this.answers.componentPath),
+    const componentName = path.basename(this.answers.componentPath);
+
+    invariant(
+      casing.isPascalCase(componentName),
+      `The component name ${componentName} must be in PascalCase.`,
     );
```

Comment:
This makes the README for the React component generator `(generator-ds/src/component/README.md)` inaccurate, as it recommends running `yo @canonical/ds:component path/to/component`, which would break this invariant.

Also, that component-specific README should be updated to remove the `--withStyles` and `--withStories` flags, as we are relying on `--help` to be authoritative parameter documentation.